File: python-ref/gvibu_ref/commands/sleep.py
"""sleep: delay for a specified number of seconds."""

import sys
import time


def run(args: list[str]) -> int:
    if not args:
        print("sleep: usage: sleep NUMBER", file=sys.stderr)
        return 2

    try:
        seconds = int(args[0])
    except ValueError:
        print(f"sleep: invalid number: {args[0]}", file=sys.stderr)
        return 2

    if seconds < 0:
        print(f"sleep: invalid number: {args[0]}", file=sys.stderr)
        return 1

    if seconds > 0:
        time.sleep(seconds)
    return 0

File: rust/src/commands/sleep.rs
/// sleep: delay for a specified number of seconds.
use std::io::Write;
use std::time::Duration;

pub fn run(stdout: &mut dyn Write, args: &[String]) -> i32 {
    let _ = stdout;
    if args.is_empty() {
        eprintln!("sleep: usage: sleep NUMBER");
        return 2;
    }

    let seconds: i64 = match args[0].parse() {
        Ok(n) => n,
        Err(_) => {
            eprintln!("sleep: invalid number: {}", args[0]);
            return 2;
        }
    };

    if seconds < 0 {
        eprintln!("sleep: invalid number: {}", args[0]);
        return 1;
    }

    if seconds > 0 {
        std::thread::sleep(Duration::from_secs(seconds as u64));
    }
    0
}
